tests/probes_test.go

```diff
@@ -69,22 +60,44 @@ var _ = Describe("[ref_id:1182]Probes", func() {
 			return vmiReady(readVmi) == v1.ConditionTrue
 		}
 
-		table.DescribeTable("should succeed", func(readinessProbe *v12.Probe, serverStarter func(vmi *v12.VirtualMachineInstance, port int)) {
-			By("Specifying a VMI with a readiness probe")
-			vmi = createReadyCirrosVMIWithReadinessProbe(virtClient, readinessProbe)
+		table.DescribeTable("should succeed", func(readinessProbe *v12.Probe, IPFamily v1.IPFamily) {
+
+			if IPFamily == v1.IPv6Protocol {
+				libnet.SkipWhenNotDualStackCluster(virtClient)
+				By("Create a support pod which will reply to kubelet's probes ...")
+				probeBackendPod, supportPodCleanupFunc := buildProbeBackendPodSpec(readinessProbe)
+				defer func() {
+					Expect(supportPodCleanupFunc()).To(Succeed(), "The support pod responding to the probes should be cleaned-up at test tear-down.")
+				}()
+
+				By("Attaching the readiness probe to an external pod server")
+				supportPodIP := libnet.GetPodIpByFamily(probeBackendPod, IPFamily)
+				Expect(supportPodIP).ToNot(BeEmpty())
+				readinessProbe = patchProbeWithIPAddr(readinessProbe, supportPodIP)
+
+				By("Specifying a VMI with a readiness probe")
+				vmi = createReadyCirrosVMIWithReadinessProbe(virtClient, readinessProbe)
+			} else {
+				By("Specifying a VMI with a readiness probe")
+				vmi = createReadyCirrosVMIWithReadinessProbe(virtClient, readinessProbe)
+
+				By("Starting the server inside the VMI")
+				serverStarter(vmi, readinessProbe, 1500)
+			}
 
 			// pod is not ready until our probe contacts the server
 			assertPodNotReady(virtClient, vmi)
 
-			By("Starting the server inside the VMI")
-			serverStarter(vmi, 1500)
-
 			By("Checking that the VMI and the pod will be marked as ready to receive traffic")
 			Eventually(isVMIReady, 60, 1).Should(Equal(true))
 			Expect(tests.PodReady(tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault))).To(Equal(v1.ConditionTrue))
 		},
-			table.Entry("[test_id:1202][posneg:positive]with working TCP probe and tcp server", tcpProbe, tests.StartTCPServer),
-			table.Entry("[test_id:1200][posneg:positive]with working HTTP probe and http server", httpProbe, tests.StartHTTPServer),
+			table.Entry("[test_id:1202][posneg:positive]with working TCP probe and tcp server,no ip family specified ", tcpProbe, *new(v1.IPFamily)),
+			table.Entry("[test_id:1202][posneg:positive]with working TCP probe and tcp server on ipv4", tcpProbe, v1.IPv4Protocol),
+			table.Entry("[test_id:1202][posneg:positive]with working TCP probe and tcp server on ipv6", tcpProbe, v1.IPv6Protocol),
+			table.Entry("[test_id:1202][posneg:positive]with working TCP probe and tcp server,no ip family specified ", httpProbe, *new(v1.IPFamily)),
```

Comment:
@alonSadan I think what you want this table entry to be called is this.
```suggestion
			table.Entry("[test_id:1202][posneg:positive]with working HTTP probe and tcp server,no ip family specified ", httpProbe, *new(v1.IPFamily)),
```

As the merge tool relies on unique naming during processing it thinks that a test has run more than once.

Reply:
Thank you! 
what a stupid mistake.
Fixed.